src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -124,17 +124,25 @@ uct_rc_mlx5_iface_poll_tx(uct_rc_mlx5_iface_common_t *iface)
     ucs_trace_poll("rc_mlx5 iface %p tx_cqe: ep %p qpn 0x%x hw_ci %d", iface, ep,
                    qp_num, hw_ci);
 
+    /* Process completions before updating resources to avoid OOO.
+     * Some resources (RDMA_READ credits) are updated in operation completion
+     * callbacks, so need to call all relevant callbacks prior dispatching
+     * arbiter. In this case ordering is preserved, because:
+     * - If nothing is sent from the completion callback, we just update
+     *   resources and dispatch arbiter.
+     * - If something needs to be sent from completion callback, but not all
+     *   needed resources available, it will be added to the pending queue
+     *   (and handled in dispatch below in the proper order).
+     */
```

Comment:
seems this comment is copy-pasted in multiple transports, maybe move it to uct_rc_iface_update_reads